portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/DataSourcesList.testcase

```diff
@@ -57,4 +57,52 @@ definition {
 		}
 	}
 
+	@description = "Bug: LRAC-10507 | Automation ID: LRAC-12349 | Test Summary: The data source list has pagination"
+	@priority = "3"
+	test PaginateDataSourceList {
+		var dataSourceList = "";
+
+		TestCase.setUpPortalInstance();
+
+		for (var n : list "1,2,3,4,5,6") {
+			ACSettings.takeTokenFromAnalyticsCloud();
+
+			ACUtils.launchDXP();
+
+			var propertyName = ACDXPSettings.connectDXPtoAnalyticsCloud();
+
+			var dataSource = StringUtil.replace("${propertyName}", " Combined Property", "");
+
+			var dataSourceList = StringUtil.add("${dataSourceList}", "${dataSource}");
+
+			ACDXPSettings.disconnectDXPFromAnalyticsCloud();
+
+			ACUtils.launchAC();
+
+			ACNavigation.goToSettings();
+		}
+
+		task ("Remove the last data source from the list to iterate just the first five") {
+			var newDataSourceList = ListUtil.newListFromString("${dataSourceList}");
```

Comment:
Hey @manuferreira, I think you don't need to initialize this new list. You couldn't remove the last data source (${dataSource}) directly from the list you use before (${dataSourceList}). What do you think?

Reply:
Thanks for noticing!

Reply:
Thanks manu 😄